commit ab2fb0c7aaeeef7e1411564663a9da041146f0d3 (refs/changes/66/966/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-11-29T08:07:46-05:00 (2 years, 10 months ago)
    
    joyent/node-cueball#54 skip AAAA lookups on machines with no v6 address

diff --git a/lib/resolver.js b/lib/resolver.js
index aa0474f..851c424 100644
--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -34,6 +34,7 @@ const mod_ipaddr = require('ipaddr.js');
 const mod_fs = require('fs');
 const mod_crypto = require('crypto');
 const mod_verror = require('verror');
+const mod_os = require('os');
 
 const FSM = mod_mooremachine.FSM;
 const EventEmitter = mod_events.EventEmitter;
@@ -465,9 +466,21 @@ CueBallDNSResolver.prototype.state_srv_error = function (S) {
 };
 
 CueBallDNSResolver.prototype.state_aaaa = function (S) {
-	this.r_srvRem = this.r_srvs.slice();
-	this.r_nextV6 = undefined;
-	S.gotoState('aaaa_next');
+	var nics = mod_os.networkInterfaces();
+	var haveV6 = false;
+	Object.keys(nics).forEach(function (k) {
+		nics[k].forEach(function (addr) {
+			if (addr.family === 'IPv6' && addr.internal !== true)
+				haveV6 = true;
+		});
+	});
+	if (haveV6) {
+		this.r_srvRem = this.r_srvs.slice();
+		this.r_nextV6 = undefined;
+		S.gotoState('aaaa_next');
+	} else {
+		S.gotoState('a');
+	}
 };
 
 CueBallDNSResolver.prototype.state_aaaa_next = function (S) {
